strawberry_django/optimizer.py

```diff
@@ -196,8 +197,10 @@ def with_prefix(self, prefix: str, *, info: GraphQLResolveInfo):
             if isinstance(p, str):
                 prefetch_related.append(f"{prefix}{LOOKUP_SEP}{p}")
             elif isinstance(p, Prefetch):
-                p.add_prefix(prefix)
-                prefetch_related.append(p)
+                # add_prefix modifies the field's prefetch object, so we copy it before
+                p_copy = deepcopy(p)
```

Comment:
Hrm, the only solution I can think right now would be to create a `Prefetch` object by hand, but then we would need to update it in case it gets changed in django.

The only question here would be: Any reason to use `deepcopy` instead of `copy`? Unless I'm missing something, a copy should be enough to fix the issue here

Reply:
Yeah, I tried that and it's also not super trivial with the internal `levels` and protected variables. Kind of a messy class.

I don't really have a good reason for deepcopy, fair point, I changed it to `copy.copy` and the test still works :)